### PR TITLE
Small fixes and refactorings

### PR DESCRIPTION
**Pull request by rasendubi**

### CODE REVIEW

File: eppoclient/initclient.go [hunk 1]
```diff
@@ -13,7 +13,7 @@ func InitClient(config Config) *EppoClient {
 	sdkParams := SDKParams{apiKey: config.ApiKey, sdkName: "go", sdkVersion: __version__}
 
 	httpClient := newHttpClient(config.BaseUrl, &http.Client{Timeout: REQUEST_TIMEOUT_SECONDS}, sdkParams)
-	configStore := newConfigurationStore(MAX_CACHE_ENTRIES)
+	configStore := newConfigurationStore()
```
Comment (pavelflux) [inline eppoclient/initclient.go:16]: why was this removed?

Comment (rasendubi) [reply eppoclient/initclient.go:16]: The cache is used incorrectly in the SDKs. It does not speed up things but rather adds overhead (to track which entries were used last).

**Review by pavelflux — APPROVED** [review]

File: eppoclient/configurationstore.go [hunk 2]
```diff
@@ -2,11 +2,10 @@ package eppoclient
 
 import (
 	"errors"
-	lru "github.com/hashicorp/golang-lru/v2"
 )
 
 type configurationStore struct {
-	cache *lru.Cache[string, experimentConfiguration]
+	configs map[string]experimentConfiguration
```
Comment (leoromanovsky) [inline eppoclient/configurationstore.go:8]: 👍

File: eppoclient/rules.go [hunk 3]
```diff
@@ -60,7 +60,7 @@ func evaluateCondition(subjectAttributes dictionary, condition condition) bool {
 		return isOneOf(subjectValue, convertToStringArray(condition.Value))
 	case "NOT_ONE_OF":
 		return isNotOneOf(subjectValue, convertToStringArray(condition.Value))
-	default:
+	case "GTE", "GT", "LTE", "LT":
```
Comment (leoromanovsky) [inline eppoclient/rules.go:63]: 👍

File: eppoclient/shard.go [hunk 4]
```diff
@@ -2,26 +2,21 @@ package eppoclient
...
 func getShard(input string, subjectShards int64) int64 {
 	hash := md5.Sum([]byte(input))
-	hashOutput := hex.EncodeToString(hash[:])
-
-	// get the first 4 bytes of the md5 hex string and parse it using base 16
-	// (8 hex characters represent 4 bytes, e.g. 0xffffffff represents the max 4-byte integer)
-	intVal, _ := strconv.ParseInt(hashOutput[0:8], 16, 0)
-
+	// Only first 4 bytes of md5 are used for the shard value.
+	intVal := int64(binary.BigEndian.Uint32(hash[:4]))
```
Comment (leoromanovsky) [inline eppoclient/shard.go:16]: nice improvement. i read them as equivalent and this is more straight-forward.

**Review by leoromanovsky — APPROVED** [review]
